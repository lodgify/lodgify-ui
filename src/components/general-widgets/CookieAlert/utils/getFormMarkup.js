import React from 'react';

import { Form } from 'collections/Form';
import { Paragraph } from 'typography/Paragraph';
import { Link } from 'elements/Link';

/**
 * @param {string} buttonText
 * @param {string} linkText
 * @param {string} linkUrl
 * @param {Function} onAccept
 * @param {string} text
 * @return {Object}
 */
export const getFormMarkup = (
  buttonText,
  linkText,
  linkUrl,
  onAccept,
  text
) => (
  <Form onSubmit={onAccept} submitButtonText={buttonText}>
    <Paragraph>{text}</Paragraph>
    <Link href={linkUrl}>{linkText}</Link>
  </Form>
);
