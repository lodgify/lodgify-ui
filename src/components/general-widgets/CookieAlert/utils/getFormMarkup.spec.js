import { getFormMarkup } from './getFormMarkup';

describe('`getFormMarkup`', () => {
  const onAccept = jest.fn();
  const buttonText = 'Le Goose';
  const text = 'eau';
  const linkText = 'Bip';
  const linkUrl = 'Bup';

  it('should render the right structure', () => {
    const actual = getFormMarkup(buttonText, linkText, linkUrl, onAccept, text);

    expect(actual).toMatchSnapshot();
  });
});
