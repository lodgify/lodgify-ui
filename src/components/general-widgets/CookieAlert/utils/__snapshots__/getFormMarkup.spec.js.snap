// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`getFormMarkup\` should render the right structure 1`] = `
<Form
  actionLink={null}
  autoComplete={null}
  errorMessage=""
  headingText={null}
  onSubmit={[MockFunction]}
  submitButtonText="Le Goose"
  successMessage=""
  validation={Object {}}
>
  <Paragraph
    size="medium"
    weight={null}
  >
    eau
  </Paragraph>
  <Link
    href="Bup"
    isPositionedRight={false}
    onClick={[Function]}
    willOpenInNewTab={false}
  >
    Bip
  </Link>
</Form>
`;
